test/core/channel/metrics_test.cc

```diff
@@ -301,194 +301,160 @@ class FakeStatsPluginBuilder {
-                   .BuildAndRegister();
-  }
-
   void TearDown() override {
-    GlobalStatsPluginRegistry::TestOnlyResetStatsPlugins();
-  }
-
-  static void SetUpTestSuite() {
-    // Initializer list.
-    uint64_counter_handle_ = GlobalInstrumentsRegistry::RegisterUInt64Counter(
-        "uint64_counter", "A simple uint64 counter.", "unit",
-        {"label_key_1", "label_key_2"},
-        {"optional_label_key_1", "optional_label_key_2"});
-    // C-style array.
-    constexpr absl::string_view kLabelKeys[] = {"label_key_1", "label_key_2"};
-    constexpr absl::string_view kOptionalLabelKeys[] = {"optional_label_key_1",
-                                                        "optional_label_key_2"};
-    double_counter_handle_ = GlobalInstrumentsRegistry::RegisterDoubleCounter(
-        "double_counter", "A simple double counter.", "unit", kLabelKeys,
-        kOptionalLabelKeys);
-    // Explicit container.
-    const std::vector<absl::string_view> kLabelKeysContainer = {"label_key_1",
-                                                                "label_key_2"};
-    const std::vector<absl::string_view> kOptionalLabelKeysContainer = {
-        "optional_label_key_1", "optional_label_key_2"};
-    uint64_histogram_handle_ =
-        GlobalInstrumentsRegistry::RegisterUInt64Histogram(
-            "uint64_histogram", "A simple uint64 histogram.", "unit",
-            kLabelKeysContainer, kOptionalLabelKeysContainer);
-    double_histogram_handle_ =
-        GlobalInstrumentsRegistry::RegisterDoubleHistogram(
-            "double_histogram", "A simple double histogram.", "unit",
-            kLabelKeysContainer, kOptionalLabelKeysContainer);
+    GlobalInstrumentsRegistry::TestOnlyResetGlobalInstrumentsRegistry();
+    GlobalStatsPluginRegistry::TestOnlyResetGlobalStatsPluginRegistry();
   }
-
- protected:
-  std::shared_ptr<FakeStatsPlugin> plugin1_;
-  std::shared_ptr<FakeStatsPlugin> plugin2_;
-  std::shared_ptr<FakeStatsPlugin> plugin3_;
-
-  static GlobalInstrumentsRegistry::GlobalUInt64CounterHandle
-      uint64_counter_handle_;
-  static GlobalInstrumentsRegistry::GlobalDoubleCounterHandle
-      double_counter_handle_;
-  static GlobalInstrumentsRegistry::GlobalUInt64HistogramHandle
-      uint64_histogram_handle_;
-  static GlobalInstrumentsRegistry::GlobalDoubleHistogramHandle
-      double_histogram_handle_;
 };
 
-GlobalInstrumentsRegistry::GlobalUInt64CounterHandle
-    MetricsTest::uint64_counter_handle_;
-GlobalInstrumentsRegistry::GlobalDoubleCounterHandle
-    MetricsTest::double_counter_handle_;
-GlobalInstrumentsRegistry::GlobalUInt64HistogramHandle
-    MetricsTest::uint64_histogram_handle_;
-GlobalInstrumentsRegistry::GlobalDoubleHistogramHandle
-    MetricsTest::double_histogram_handle_;
-
 TEST_F(MetricsTest, UInt64Counter) {
+  auto uint64_counter_handle = GlobalInstrumentsRegistry::RegisterUInt64Counter(
+      "uint64_counter", "A simple uint64 counter.", "unit",
+      {"label_key_1", "label_key_2"},
```

Comment:
These label-list parameters should use the constants you're defining immediately below.

Same thing for all tests.

Reply:
Done.